javascript/react/basics-i/component-lifecycle-methods.md

```diff
@@ -26,18 +26,19 @@ aspects:
 
 ---
 
-# Component **lifecycle methods**
+# Component lifecycle methods
 
 ---
 ## Content
 
-As discussed before, **lifecycle methods** can be used to target a specific point in the *life* of a `component`.
+As discussed before, **lifecycle methods** can be used to target a specific point in the *life* of a component.
 
 We can split them into three broader categories by the *type of event* that will trigger them:
 
 ### 1. Mounting
 
-- `constructor(props)` - the first method called in the lifecycle, **before mounting**. It's most commonly used to set the initial `state` and to `bind` methods. If used, it must include `super(props)` as first call:
+- `constructor(props)` 
+  - the first method called in the lifecycle, **before mounting**. It's most commonly used to set the initial `state` and to `bind` methods. If used, it must include `super(props)` as the first call:
```

Comment:
Add a footnote for why it must include `super(props)`.

Article to add: https://overreacted.io/why-do-we-write-super-props/

Reply:
Added the following footnote:

In React, `super` refers to the `React.Component` class constructor implementation. Because of the way JavaScript works, you can't use this in your `constructor` method until after you call `super`. The `props` in `super(props)` is called so that the base `React.Component` constructor can initialize `this.props`. If you want to learn more about this make sure to read [this article](https://overreacted.io/why-do-we-write-super-props/).

Reply:
Nice. 

What do you think of this? Is it too long?

---

In JavaScript classes, the constructor is commonly used to setup the initial data of the class.

```js
class Enki {
  constructor() {
    this.greeting = 'Hello 👋';
  }
}
```

Classes can also extend behavior from other classes. 

```js
class Example extends Enki {

}
```

For the `Example` class to have access to the `greeting` from the `Enki` class, it needs to tell the `Enki` class to run its constructor and initialize the `greeting`. This is achieved with the `super` keyword.

```js
class Example extends Enki {
  constructor() {
    super(); // tell Enki to run its constructor
  }
}
```

In React, we have to call `super(props)` in a class component to tell `React.Component` to run its setup, such as initializing props.

For more info, [check out this article](https://overreacted.io/why-do-we-write-super-props/).